lib/fir/fir-basicblock.h

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2018, NVIDIA CORPORATION.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef FORTRAN_IR_BASICBLOCK_H_
+#define FORTRAN_IR_BASICBLOCK_H_
+
+namespace Fortran::IR {
+class BasicBlock;
+class Statement;
+struct Stmt_impl;
+}
+
+#include "fir-region.h"
+#include "mixin.h"
+#include "statements.h"
+#include "llvm/ADT/ilist.h"
+#include <iostream>
```

Comment:
We can't include iostream, at least not permanently.

Reply:
Why not?

Reply:
[](https://llvm.org/docs/CodingStandards.html#include-iostream-is-forbidden)

It's a special case of the LLVM rule against statically allocated objects with constructors being present in libraries.